scripted-tests/scala3/opaque-type/src/main/scala/Hello.scala

```diff
@@ -0,0 +1,13 @@
+import scalanative.libc.stdio
+import scalanative.unsafe.*
+import scalanative.unsigned.*
+
+opaque type cmark_event_type = CUnsignedInt
+object cmark_event_type:
+  inline def define(inline a: Long): cmark_event_type = a.toUInt
+  val CMARK_EVENT_NONE = define(0)
+  
+@main def hello(): Unit =
+  val evtype = cmark_event_type.CMARK_EVENT_NONE
+  stdio.printf(c"bla: %s, hello: %d", evtype)
+  
```

Comment:
Instead of expensive scripted tests, let's define it as quick unit test in nscplugin, for example in this test https://github.com/scala-native/scala-native/blob/8df8dce73d8bcbdf076f75fb0b05c45bc4a84f6c/nscplugin/src/test/scala-3/scala/scalanative/NIRCompilerTest3.scala#L12-L13 

We only need to ensure that is compiles, we don't require linking

Reply:
Thanks! done in https://github.com/scala-native/scala-native/pull/3712/commits/1a3dbcc4dd232ede14397f789dea94a87673fafa
(I added clib.cp to `Test/  nscplugin / javaOptions` since I couldn't find a way to reproduce the compilation crash without using `clib.stdio` 🤔

Reply:
I think you might have problems with that, becouse there are 2 definitions of printf. One is defined as extension here: https://github.com/scala-native/scala-native/blob/ba2bd19e0ccc1d3ec6db97a6249aa397a4f7e308/clib/src/main/scala/scala/scalanative/libc/package.scala#L5-L8 it's a legacy support from before introduction of better varargs in extern  
The second one is standard extern binding https://github.com/scala-native/scala-native/blob/ba2bd19e0ccc1d3ec6db97a6249aa397a4f7e308/clib/src/main/scala/scala/scalanative/libc/stdio.scala#L587-L589 

I belive it could have been coused by the clash of the two. It would be great if we could get rid of the dependency. I'd like to remove legacy version in the future.

Reply:
Thanks @WojciechMazur!
I'm not sure the duplication of `printf` is the problem, I could reproduce it without `clib` https://github.com/scala-native/scala-native/pull/3712/commits/74fbdddc1e59697f5a76a524b3a4107ba812513c

It seems like we hit the `ensureUnboxed` -> `isPrimitiveType` when the argument is for extern method's varargs (e.g. `printf`'s `vargs`)
https://github.com/scala-native/scala-native/blob/74fbdddc1e59697f5a76a524b3a4107ba812513c/nscplugin/src/main/scala-3/scala/scalanative/nscplugin/NirGenExpr.scala#L1722-L1723 

With something like `def test(t: Any*) = extern`, and if I pass the opaque typed value to the `test` method, we could break the compiler :+1: